tsdb/wlog/watcher.go

```diff
@@ -488,6 +491,7 @@ func (w *Watcher) readSegment(r *LiveReader, segmentNum int, tail bool) error {
 		histogramsToSend      []record.RefHistogramSample
 		floatHistograms       []record.RefFloatHistogramSample
 		floatHistogramsToSend []record.RefFloatHistogramSample
+		metadata              []record.RefMetadata
```

Comment:
should we copy the same `samples` and `samplesToSend` pattern?

Reply:
ah you're right :facepalm: yes we should treat this the same as the series records, and just always send them

the queue manager is responsible for knowing what to do with the metadata it receives, see my comment there

Reply:
I see! With the new `StoreMetadata` approach, we always append metadata in Append/AppendExemplar/AppendHistogram if we find it in the cache.

Resolving this conversation